chore: remove unnecessary code, adding the IoT retirement in README (#511)
googleapis/python-iot: README.rst

> :**NOTE**: **This github repository is archived. Google Cloud IoT Core is being retired in August 2023. For details, see https://cloud.google.com/iot-core. **

Reviewer: ```suggestion
:**NOTE**: **This github repository is archived. Google Cloud IoT Core is retired as of August 2023. For details, see https://cloud.google.com/iot-core. **
```
